packages/components/src/components/input-date/types.ts

```diff
@@ -1,23 +1,48 @@
 import { Generic } from '@a11y-ui/core';
+import type { ButtonProps } from '../../types/button-link';
+import type { KoliBriHorizontalIcon } from '../../types/icon';
 import { InputDateType, OptionalInputProps } from '../../types/input/control/number';
 import { Iso8601 } from '../../types/input/iso8601';
-import { InputTypeOnDefault } from '../../types/input/types';
+import type { InputTypeOnDefault, InputTypeOnOff } from '../../types/input/types';
 import { InputRequiredProps } from '../input-text/types';
 
 type RequiredProps = InputRequiredProps;
 type OptionalProps = OptionalInputProps<Iso8601 | Date> & { type: InputDateType };
 
-type RequiredStates = unknown;
+type RequiredStates = {
+	autoComplete: InputTypeOnOff;
+	hasValue: boolean;
+	id: string;
+	list: string[];
+	type: InputDateType;
+};
 
 type OptionalStates = {
-	on: InputTypeOnDefault;
+	accessKey: string;
+	alert: boolean;
+	disabled: boolean;
+	error: string;
+	hideLabel: boolean;
+	hint: string;
+	icon: KoliBriHorizontalIcon;
 	max: Iso8601;
 	min: Iso8601;
-	value: Iso8601 | null;
+	name: string;
+	on: InputTypeOnDefault;
+	placeholder: string;
+	readOnly: boolean;
+	required: boolean;
+	smartButton: ButtonProps;
+	step: number;
+	tabIndex: number;
+	touched: boolean;
+	value: Iso8601;
```

Comment:
The value can also be null. This is the case when the input-field is empty.

Reply:
Empty value would mean an empty string, or not? Looking at the other "input" components or even textarea there is no `null`, but from the type it could be a `undefined`. Shouldn't the value be a default empty string used like this: `this.state_value || ''` ?